### ISSUE TITLE
svelte-check fails with real codebase

### ISSUE THREAD

**Issue by rallets** [johnsoncodehk/typescript-native-bridge#46]
Hi, first, thanks for your work, really appreciated to be able to use/test ts7 before svelte team close the gap with the new developer API.
I was trying to use your bridge but svelte-check fails with thousand of error like:
`Error: Cannot find name '__sveltets_2_ensureComponent'. (ts)`
`Error: Cannot find name '__sveltets_2_any'. (ts)`
`Error: Cannot find name '__sveltets_2_ensureArray'. (ts)`
etc
As I don't know enough knowledge of the compiler internals, I asked Claude Sonnet 4.6 to investigate if it was an issue from my side, as in the readme.md it mention that svelte-check is upported (fixture project), and it says:

Here's the TNB fixture's entire Svelte test:
``` svelte
<!-- src/App.svelte — the ENTIRE fixture -->
<script lang="ts">
  import { double } from './util';
  let { count }: { count: number } = $props();
  const wrong: string = double(count);
</script>
<p>{double(count)}</p>
```

The compatibility check only asserts two things:

1. No `Cannot find name 'svelteHTML'` false positive
2. Exactly 1 error is found (the deliberate `string = number` assignment)
That's it. One file, one prop, one import.

Why your project breaks while the fixture passes:

`svelte2tsx` generates different virtual TypeScript code depending on what Svelte features you use. `__sveltets_2_any` is a helper type defined in `svelte2tsx/svelte-shims-v4.d.ts` — it appears in the generated virtual code when svelte2tsx encounters more complex patterns that the fixture simply doesn't have, such as:

- Snippets (`{#snippet ...}`)
- `{@render ...}` tags
- Complex generic component props
- Certain template expression patterns
- More intricate `$props()` destructuring
- 
The fixture's single `$props()` with a plain `{ count: number }` doesn't trigger the code paths in svelte2tsx that emit `__sveltets_2_any`. Your 100+ real components do.

So TNB's "✅ svelte-check compatible" claim is technically true for that single-file, single-prop Svelte 5 component — but it's not a representative test of a real Svelte 5 codebase. The compatibility gap is between what the fixture exercises and what your code actually generates.

**Comment by johnsoncodehk** [comment]
Fixed on master by 9bab476: `updateAPIRootFiles` dropped the svelte2tsx shim roots (`svelte-shims-v4.d.ts`) on the next snapshot rebuild, leaving every global `__sveltets_2_*` helper and `svelteHTML` unresolved — exactly the errors in your report.

Reproduced on the published bridge.7 package with fixtures exercising component renders (`ensureComponent`), `{#each}` (`ensureArray`) and `{#snippet}`/`{@render}` (`ensureSnippet`); master is byte-identical to stock typescript@6.0.3 on the same fixtures. The svelte gate now covers these patterns (7963c44). Ships with bridge.8.
